src/rest/src/main/java/org/geoserver/rest/converters/StyleWriterConverter.java

```diff
@@ -40,7 +40,7 @@ public StyleWriterConverter(String mimeType, Version version, StyleHandler handl
 
     @Override
     protected boolean supports(Class<?> clazz) {
-        return Style.class.isAssignableFrom(clazz) || StyleInfo.class.isAssignableFrom(clazz);
+        return RestWrapper.class.isAssignableFrom(clazz) || Style.class.isAssignableFrom(clazz) || StyleInfo.class.isAssignableFrom(clazz);
```

Comment:
The code ended up here with a RestWrapper, I did not like it either but... we need an API that does the following:

- /styles/foo.html -> html, so rest wrapper
- /styles/foo.xml -> xml, so StyleInfo
- /styles/foo.sld -> sld, for native sld, should be a dump of the file, but for non sld, should be a conversion
- /styles/foo.xyz -> fow a dynamic xyz that a style handler handles, should be dumping the original style

Maybe have a get path that explicitly handles html, one for xml, and a final one returning a StyleInfo object (unwrapped) for everything else? Assuming MVC allows this setup that is.